### PR TITLE
Catch untracked files by node test generation in CI

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

- Catch untrack files by node test generation in a CI. I think single CI should be sufficient.
- For consistency, set ORT_MAX_ONNX_OPSET_SUPPORTED_VERSION=19 in all CIs.
- Update missing Reduce backend node tests.

### Motivation and Context

Inspired by https://github.com/onnx/onnx/pull/5614#pullrequestreview-1640344022. Currently CI does not catch those untrack files generated by node test generation. Having such a test can prevent missing node backend tests in the future.

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

**Comment by jcwchen** [comment]
It doesn't throw an error in CI somehow... I will take a deeper look.

File: .azure-pipelines/MacOS-CI.yml [hunk 1]
```diff
@@ -119,4 +119,9 @@ jobs:
         exit 1
       fi
 
+      git ls-files --others --error-unmatch --exclude-standard
+      if [ $? -ne 0 ]; then
+        echo "git diff for test generation, because there might be some untrack files. Please upload them as well"
```
Comment (gramalingam) [inline .azure-pipelines/MacOS-CI.yml:124]: nit: may be improve the message a bit: "Some test-generated files not included in PR. Did you forget to add any test-generated files?"

**Review by gramalingam — APPROVED** [review]

File: .azure-pipelines/MacOS-CI.yml (hunk 1 above)
Comment (jcwchen) [reply .azure-pipelines/MacOS-CI.yml]: Looks better. I have updated it. PTAL. Thanks!

**Comment by jcwchen** [comment]
Now the CI can catch such an untracked files correctly.

**Review by gramalingam — APPROVED** [review]

**Comment by justinchuby** [comment]
Reduce cases need to be generated too

**Comment by jcwchen** [comment]
> Reduce cases need to be generated too

I updated them in this PR and now the CI passed. PTAL. Thanks!

**Comment by jcwchen** [comment]
@liqunfu I guess this PR needs to be cherry-picked? It adds missing backend node tests.

**Pull request merged**